x-pack/platform/plugins/shared/lens/public/datasources/form_based/form_based.tsx

```diff
@@ -745,23 +1050,113 @@ export function getFormBasedDatasource({
           }
           return null;
         },
-        hasDefaultTimeField: () => Boolean(indexPatterns[layer.indexPatternId].timeFieldName),
+        hasDefaultTimeField: () =>
+          Boolean(layer.indexPatternId && indexPatterns[layer.indexPatternId].timeFieldName),
       };
     },
     getDatasourceSuggestionsForField(state, draggedField, filterLayers, indexPatterns) {
+      const layers = Object.values(state.layers);
+      if (layers?.[0] && isTextBasedLayer(layers[0])) {
+        const query = layers[0].query;
+        const fieldList = query ? getColumnsFromCache(query) : [];
+        const field = fieldList?.find((f) => f.id === (draggedField as TextBasedField).id);
+        if (!field) return [];
+        return Object.entries(state.layers)?.map((entry) => {
+          const [id] = entry;
+          const layer = entry[1] as TextBasedLayer;
+          const newId = generateId();
+          const newColumn = {
+            columnId: newId,
+            fieldName: field?.name ?? '',
+            meta: field?.meta,
+          };
+          return {
+            state: {
+              ...state,
+              layers: {
+                ...state.layers,
+                [id]: {
+                  ...layer,
+                  columns: [...layer.columns, newColumn],
+                },
+              },
+            },
+            table: {
+              changeType: 'initial' as TableChangeType,
+              isMultiRow: false,
+              layerId: id,
+              columns: [
+                ...layer.columns?.map((f) => {
+                  return {
+                    columnId: f.columnId,
+                    operation: {
+                      dataType: f?.meta?.type as DataType,
+                      label: f.fieldName,
+                      isBucketed: Boolean(isNotNumeric(f)),
+                    },
+                  };
+                }),
+                {
+                  columnId: newId,
+                  operation: {
+                    dataType: field?.meta?.type as DataType,
+                    label: field?.name ?? '',
+                    isBucketed: Boolean(isNotNumeric(field)),
```

Comment:
nit: does it need `Boolean(...)` wrapping?

Reply:
this was copied over from text_based datasource, i would prefer to keep the code unchanged as much as possible